### DISCUSSION TITLE
Cannot replace new version due to nerdctl file being used

### DISCUSSION THREAD

**Discussion by DesistDaydream** [containerd/nerdctl#1949] [Q&A] [answered]
### Description

The nerdctl file cannot be replaced, and the error is as follows:
`cp: cannot create regular file '/usr/local/bin/nerdctl': Text file busy`

```bash
~]# for i in $(lsof -t /usr/local/bin/nerdctl); do cat /proc/${i}/cmdline && echo ; done
/usr/local/bin/nerdctl_NERDCTL_INTERNAL_LOGGING/var/lib/nerdctl/1935db59
/usr/local/bin/nerdctl_NERDCTL_INTERNAL_LOGGING/var/lib/nerdctl/1935db59
/usr/local/bin/nerdctl_NERDCTL_INTERNAL_LOGGING/var/lib/nerdctl/1935db59
/usr/local/bin/nerdctl_NERDCTL_INTERNAL_LOGGING/var/lib/nerdctl/1935db59
```

### Steps to reproduce the issue

1.  wget https://github.com/containerd/nerdctl/releases/download/v1.1.0/nerdctl-1.1.0-linux-amd64.tar.gz
2. tar -zxvf nerdctl-1.1.0-linux-amd64.tar.gz
3. cp nerdctl /usr/local/bin/

The nerdctl file cannot be replaced, and the error is as follows:
`cp: cannot create regular file '/usr/local/bin/nerdctl': Text file busy`

### Describe the results you received and expected

The CLI should not be used by any process for a long time

We upgrade CLI version, should not stop all containers

### What version of nerdctl are you using?

nerdctl version 1.1.0

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

Client:
 Namespace:	default
 Debug Mode:	false

Server:
 Server Version: v1.6.2
 Storage Driver: overlayfs
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Log: fluentd journald json-file syslog
  Storage: aufs native overlayfs
 Security Options:
  apparmor
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.4.0-125-generic
 Operating System: Ubuntu 20.04.5 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 7.771GiB
 Name: hw-cloud-xngy-jump-server-linux-2
 ID: 71ef35e4-36f9-40fa-a1cb-8f8af17fb67e

WARNING: No swap limit support

**Comment by AkihiroSuda** [comment]
Please stop all containers before upgrading

**Comment by DesistDaydream** [comment]
> Please stop all containers before upgrading

Can nerdctl not be used in a production environment?

**Comment by AkihiroSuda** [comment]
> > Please stop all containers before upgrading
> 
> Can nerdctl not be used in a production environment?

You can use in production, and you have to stop containers before upgrading.

**Comment by DesistDaydream** [comment]
The production environment cannot stop the container at will.

A CLI should not occupy itself all the time

**Comment by AkihiroSuda** [comment, marked as answer]
`cp -f` seems to work.
Or `rm` before `cp`.

It's still recommended to stop containers before upgrading though.
